### ISSUE TITLE
bug: incorrect assessment of unreachable code

### ISSUE THREAD

**Issue by kitsonk** [denoland/deno_lint#318]
The following code:

```ts
function normalize(type: string): string | undefined {
  switch (type) {
    case "urlencoded":
      return "application/x-www-form-urlencoded";
    case "multipart":
      return "multipart/*";
  }

  if (type[0] === "+") {
    return `*/*${type}`;
  }

  return type.includes("/") ? type : lookup(type);
}
```

Results in the following incorrect lint violation:

```
(no-unreachable) This statement is unreachable
  if (type[0] === "+") {
  ^^^^^^^^^^^^^^^^^^^^^^
    return `*/*${type}`;
^^^^^^^^^^^^^^^^^^^^^^^^
  }
^^^
    at /example.ts:9:2
```

**Comment by bartlomieju** [comment]
CC @kdy1

**Comment by Xernisfy** [comment]
An unconditional return statement inside of a class constructor also leads to a "unreachable" linting error:

```ts
class Class {
  constructor() {
    return this;
  }
}

console.log("unreachable???");
```
```
(no-unreachable) This statement is unreachable
console.log("unreachable???");
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    at \\?\C:\Users\akn\OneDrive - EPG\Synced\TMP\Demo3DWebClient\local\deno-lint.ts:7:0

Found 1 problem
Checked 1 file
```

A conditional return however does not produce an error:
```ts
class Class {
  constructor() {
    if (Deno) return this;
  }
}

console.log("unreachable???");
```

**Comment by magurotuna** [comment]
@Xernisfy 
Thanks for the clear report. Seems like that should be a bug. Could you open a new issue for that?
